concurrency-limits-core/src/test/java/com/netflix/concurrency/limits/strategy/SimpleStrategyTest.java

```diff
@@ -52,4 +58,28 @@ public void acquireAndRelease() {
         Assert.assertTrue(strategy.tryAcquire(null).isAcquired());
         Assert.assertEquals(1, strategy.getBusyCount());
     }
+
+    @Test
+    public void tryAcquireWithMultipleThreads() throws Exception {
```

Comment:
I spotted the same bug and was in the process of fixing it before seeing this PR :)

What do you think about this test? It removes the need to add the acquireBarrier in the SimpleStrategy, and seems to reproduce the race condition reliably.

```
@Test
    public void tryAcquireWithMultipleThreads() throws InterruptedException {
        final SimpleStrategy<Void> strategy = new SimpleStrategy<>();
        strategy.setLimit(1); // just in case defaults change

        // A consumer loops doing 1M acquires and releasing
        final Runnable consumer = () -> {
            for (int i = 0; i < 1000000; i++) {
                final Token t = strategy.tryAcquire(null);
                if (t.isAcquired()) {
                    // if acquired, the busy count MUST be exactly the limit (1)
                    assertEquals(strategy.getLimit(), strategy.getBusyCount());
                    t.release();
                }
            }
        };

        ExecutorService executor = Executors.newFixedThreadPool(20);
        for (int i = 0; i < 1000; i++) {
            executor.execute(consumer);
        }
        executor.shutdown();
        executor.awaitTermination(10, TimeUnit.SECONDS);
    }
```